Nutrialysis/Managers/CoreDataManager.swift

```diff
@@ -0,0 +1,123 @@
+//
+//  CoreDataManager.swift
+//  Nutrialysis
+//
+//  Created by Mohamed Sayed on 28.04.24.
+//
+
+import Foundation
+import CoreData
+
+final class CoreDataManager: ObservableObject {
+    private let container: NSPersistentContainer
+    private var isFoodAdded = false
+    
+    @Published var foodEntities = [FoodEntity]()
+    @Published var consumedDates = [String]()
+    @Published var caloriesPerDay = [String]()
+    
+    init() {
+        container = NSPersistentContainer(name: foodContainer)
+        container.loadPersistentStores { (description, error) in
+            guard let error = error else {
+                return
+            }
+            
+            print("Error Loading CoreData: \(error.localizedDescription)")
+        }
+        
+        fetchFood()
+    }
+    
+    func addNewFoodEntity(withfoodName name: String, calories: Int, quantity: Int, measure: String, photo: URL, completion: @escaping (Bool) -> Void) {
+        let foodEntity = FoodEntity(context: container.viewContext)
+        foodEntity.name = name
+        foodEntity.calories = Int16(calories)
```

Comment:
Why casting to Int16?

Reply:
Core Data Entity doesn't support Int.
Either Int16 or Int32 or Int64 and all of them require casting